pkg/agent/agentclient/client.go

```diff
@@ -171,7 +177,11 @@ func (a *AgentClient) Serve(stopCh <-chan struct{}) {
 			klog.Infof("received DATA(id=%d)", data.ConnectID)
 			klog.Infof("[tracing] %v", data)
 
-			if ctx, ok := a.connContext[data.ConnectID]; ok {
+			a.connLock.RLock()
```

Comment:
can we somehow reduce duplicate code by putting the wrapper code in a function? 
```suggestion
			a.wrapLocks(connContext)
```